tests/array_split.py

```diff
@@ -22,127 +22,110 @@
+    for shape in input_shapes:
+        a = np.random.randint(low=0, high=100, size=shape)
+        for axis in range(a.ndim):
+            input_arr = []
+            even_div = None  # a.shape[axis]
+            uneven_div = None  # a.shape[axis]
+            for div in range(1, (int)(math.sqrt(a.shape[axis]) + 1)):
+                if a.shape[axis] % div == 0:
+                    even_div = div
+                else:
+                    uneven_div = div
+                if even_div is not None and uneven_div is not None:
+                    break
+            if even_div is None:
+                even_div = a.shape[axis] if (a.shape[axis] >= 1) else 1
+            # divisible integer
+            input_arr.append(even_div)
+            # indivisble integer
+            if even_div != uneven_div and uneven_div is not None:
+                input_arr.append(uneven_div)
+            # integer larger than shape[axis]
+            input_arr.append(a.shape[axis] + np.random.randint(1, 10))
+            # indices array which has points
+            # within the target dimension of the src array
+            if a.shape[axis] > 1:
+                input_arr.append(list(range(1, a.shape[axis], even_div)))
+            # indices array which has points
+            # out of the target dimension of the src array
+            input_arr.append(
+                list(
+                    range(
+                        0,
+                        a.shape[axis] + even_div * np.random.randint(1, 10),
+                        even_div,
                     )
-                    input_arr.append(
-                        list(
-                            range(
-                                a.shape[axis]
-                                + even_div * np.random.randint(1, 10),
-                                0,
-                                -even_div,
-                            )
-                        )
+                )
+            )
+            input_arr.append(
+                list(
+                    range(
+                        a.shape[axis] + even_div * np.random.randint(1, 10),
+                        0,
+                        -even_div,
                     )
+                )
+            )
 
-                    for routine, input_opt in itertools.product(
-                        test_routine, input_arr
-                    ):
-                        # test divisible integer or indices
-                        print_msg = (
-                            f"np.{routine}({a.shape}, {input_opt}" f", {axis})"
-                        )
-                        # Check if both impls produce the error
-                        # for non-viable options
-                        b = getattr(np, routine)(a, input_opt, axis)
-                        c = getattr(num, routine)(a, input_opt, axis)
-                        is_equal = True
-                        err_arr = [b, c]
+            for routine, input_opt in itertools.product(
```

Comment:
again, what's the point of doing `product` if this test only tests `array_split`?

Reply:
It used to have `split` w/ error checking codes to see if both cunumeric and numpy produces the same error for non-viable options. I'll delete this